atomic_reactor/utils/cachi2.py

```diff
@@ -68,3 +70,74 @@ def convert_SBOM_to_ICM(sbom: Dict[str, Any]) -> Dict[str, Any]:
         {"purl": comp["purl"]} for comp in sbom["components"]  # type: ignore
     ]
     return icm
+
+
+def gen_dependency_from_sbom_component(sbom_dep: Dict[str, Any]) -> Dict[str, Optional[str]]:
+    """Generate a single request.json dependency from a SBOM component
+
+    Dependency type is derived from purl
+    """
+    # we need to detect type from purl, this is just heuristics,
+    # we cannot reliably construct type from purl
+    heuristic_type = "unknown"
+    purl = PackageURL.from_string(sbom_dep["purl"])
+    purl_type_matchers: tuple[tuple[Callable[[PackageURL], bool], str], ...] = (
+        (lambda p: p.type == "golang" and p.qualifiers.get("type", "") == "module", "gomod"),
+        (lambda p: p.type == "golang", "go-package"),
+        (lambda p: p.type == "npm", "npm"),
+        (lambda p: p.type == "pypi", "pip"),
+        (lambda p: p.type == "rpm", "rpm"),
+        (lambda p: p.type == "gem", "rubygems"),
+        (lambda p: p.type == "golang", "go-package"),
+        # TODO: cargo
+    )
+
+    for matcher, request_type in purl_type_matchers:
+        if matcher(purl):
+            heuristic_type = request_type
+            break
+
+    version = (
+        # for non-registry dependencies cachito uses URL as version
+        purl.qualifiers.get("vcs_url") or
+        purl.qualifiers.get("download_url") or
+        # for local dependencies Cachito uses path as version
+        (f"./{purl.subpath}" if purl.subpath and purl.type == "golang" else None) or
+        (f"file:{purl.subpath}" if purl.subpath and purl.type != "golang" else None) or
+        # version is mainly for dependencies from pkg registries
+        sbom_dep.get("version")
+        # returns None if version cannot be determined
+    )
+
+    return {
+        "name": sbom_dep["name"],
+        "replaces": None,  # it's always None, replacements aren't supported by cachi2
+        "type": heuristic_type,
+        "version": version,
+        # "dev": True  # TODO
```

Comment:
@chmeliik can we somehow determine this?

Reply:
For npm packages (and maybe yarn in the future) yes, from the [cdx:npm:package:bundled](https://github.com/containerbuildsystem/cachi2/blob/ae89bbab309468d3db4bc7ab3c9f88356c46b4d0/cachi2/core/models/property_semantics.py#L80) property

For others no

Reply:
added for NPM